src/server/ApiDispatchers.cpp

```diff
@@ -192,14 +191,10 @@ constexpr T saturate(auto val)
     size_t cbWritten;
     LOG_IF_FAILED(SizeTMult(outEvents.size(), sizeof(INPUT_RECORD), &cbWritten));
 
-    if (waiter)
+    if (hr == CONSOLE_STATUS_WAIT)
     {
-        hr = ConsoleWaitQueue::s_CreateWait(m, waiter.release());
-        if (SUCCEEDED(hr))
-        {
-            *pbReplyPending = TRUE;
-            hr = CONSOLE_STATUS_WAIT;
-        }
+        hr = S_OK;

```

Comment:
wait, who handles this hr? It doesn't get up to the calling application, does it?

Reply:
okay nevermind! i finally see it - all of the returns for CONSOLE_STATUS_WAIT inside host get transformed into S_OK right here. you can ignore my prior comments and write a poem about whatever the heck you want

Reply:
Sorry, yes, I was interrupted as I was writing comments. I was very confused about reading this particular bit of code, because the other 2 places return `S_OK` in this case. I later found that the returned `hr` is actually completely ignored `pbReplyPending` is true. Still, better be consistent.